packages/paypal-commerce-integration/src/paypal-commerce-alternative-methods/paypal-commerce-ratepay-payment-strategy.ts

```diff
@@ -100,42 +110,55 @@ export default class PaypalCommerceRatepayPaymentStrategy implements PaymentStra
             );
         }
 
-        try {
-            const orderId = await this.paypalCommerceIntegrationService.createOrder(
-                'paypalcommercealternativemethodscheckout',
-                { metadataId: this.guid },
-            );
-
-            const { ratepayBirthDate, ratepayPhoneNumber, ratepayPhoneCountryCode } =
-                getFieldsValues();
-
-            const paymentData = {
-                formattedPayload: {
-                    vault_payment_instrument: null,
-                    set_as_default_stored_instrument: null,
-                    device_info: null,
-                    method_id: payment.methodId,
-                    rate_pay: {
-                        birth_date: this.normalizeDate(ratepayBirthDate),
-                        phone: {
-                            national_number: ratepayPhoneNumber,
-                            country_code: ratepayPhoneCountryCode,
+        return new Promise((resolve, reject) => {
```

Comment:
Is it possible to make this part using `async await` instead of `new Promise`?

Reply:
We can't use async await in execute, for some reason it transfers us immediately to cart page and doesn't wait for async await code to execute, but Promise flow works fine. We had same issue with Braintree local methods